### PR TITLE
Refactor shop items

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/actions/RideCreateAction.hpp [hunk 1]
```diff
@@ -201,32 +201,37 @@ DEFINE_GAME_ACTION(RideCreateAction, GAME_COMMAND_CREATE_RIDE, RideCreateGameAct
         ride->num_customers_timeout = 0;
         ride->chairlift_bullwheel_rotation = 0;
 
-        ride->price = 0;
-        ride->price_secondary = 0;
+        for (auto i = 0; i < NUM_SHOP_ITEMS_PER_RIDE; i++)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/RideCreateAction.hpp:204]: Use a ranged for

File: src/openrct2/management/Marketing.cpp [hunk 2]
```diff
@@ -196,10 +196,12 @@ bool marketing_is_campaign_type_applicable(int32_t campaignType)
                 auto rideEntry = ride.GetRideEntry();
                 if (rideEntry != nullptr)
                 {
-                    if (shop_item_is_food_or_drink(rideEntry->shop_item)
-                        || shop_item_is_food_or_drink(rideEntry->shop_item_secondary))
+                    for (auto i = 0; i < NUM_SHOP_ITEMS_PER_RIDE; i++)
```
Comment (duncanspumpkin) [inline src/openrct2/management/Marketing.cpp:199]: Use a ranged for

File: src/openrct2/object/RideObject.cpp [hunk 3]
```diff
@@ -549,8 +549,10 @@ void RideObject::ReadJson(IReadObjectContext* context, const json_t* root)
     _legacyType.max_height = ObjectJsonHelpers::GetInteger(properties, "maxHeight");
 
     // This needs to be set for both shops/facilities _and_ regular rides.
-    _legacyType.shop_item = SHOP_ITEM_NONE;
-    _legacyType.shop_item_secondary = SHOP_ITEM_NONE;
+    for (auto i = 0; i < NUM_SHOP_ITEMS_PER_RIDE; i++)
```
Comment (duncanspumpkin) [inline src/openrct2/object/RideObject.cpp:552]: Ranged for

File: src/openrct2/ride/ShopItem.cpp [hunk 4]
```diff
@@ -87,139 +87,55 @@ money32 shop_item_get_common_price(Ride* forRide, int32_t shopItem)
...
+bool shop_item_has_common_price(int32_t shopItem)
 {
-    return (
-        shopItem == SHOP_ITEM_PHOTO || shopItem == SHOP_ITEM_PHOTO2 || shopItem == SHOP_ITEM_PHOTO3
-        || shopItem == SHOP_ITEM_PHOTO4);
+    return (gSamePriceThroughoutPark & (1ULL << shopItem)) != 0;
 }
 
-bool shop_item_has_common_price(int32_t shopItem)
+bool ShopItemDescriptor::HasFlag(uint16_t flag) const
```
Comment (duncanspumpkin) [inline src/openrct2/ride/ShopItem.cpp:113]: flag can be const in this

File: src/openrct2/ride/ShopItem.cpp [hunk 5]
```diff
@@ -87,139 +87,55 @@ money32 shop_item_get_common_price(Ride* forRide, int32_t shopItem)
...
+                return ride.price[1];
             }
         }
     }
 
     return MONEY32_UNDEFINED;
 }
 
-bool shop_item_is_photo(int32_t shopItem)
+bool shop_item_has_common_price(int32_t shopItem)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/ShopItem.cpp:108]: shopItem can be const
Why not make this a member function?

Comment (Gymnasiast) [reply src/openrct2/ride/ShopItem.cpp:108]: Because shop items do not know their own ID.

**Pull request merged**